jsforce/jsforce-website
# publish website via GHA

This PR adds a workflow to build and publish the jsforce website on each push to `master`.
The build is pushed to the `site` branch, then GH pages do the rest.

The site was built using `docpad`, which was deprecated a few years ago:
https://docpad.bevry.me/

we can't update to latest version because of plugins, trying to build the site on master would pass/fail intermittently on recent node versions (might be also how deps were installed?). With the changes here I tried a few times and was able to consistently get good builds.

### Testing

locally

1. checkout this PR (note it comes from my fork: https://github.com/cristiand391/jsforce-website/tree/cd/gha-build-site)
2. install deps: `yarn install`
3. build site: `NODE_ENV=production yarn docpad generate --out out`
4. serve: `yarn docpad-server`, then open http://localhost:9778/

CI:

workflow run from last week (note: I messed up with the `master` branch in my fork and restored it today, this ran from the "previous" master"
https://github.com/cristiand391/jsforce-website/actions/runs/8741378549/job/23987261125

that published to https://cristiand391.github.io/jsforce-website/

it's broken because it's serves under `cristiand391.github.io`,  once this PR merges it will be under `jsforce.github.io` so imports should be work again.
https://stackoverflow.com/questions/16316311/github-pages-and-relative-paths

@W-15508326@
